samples/seat_massager/consumer/src/main.rs

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+mod consumer_impl;
+
+use digital_twin_model::{sdv_v1 as sdv, Metadata};
+use env_logger::{Builder, Target};
+use log::{debug, info, LevelFilter, warn};
+use samples_common::constants::{digital_twin_operation, digital_twin_protocol};
+use samples_common::utils::{discover_digital_twin_provider_using_ibeji, retrieve_invehicle_digital_twin_url};
+use samples_common::consumer_config;
+use samples_protobuf_data_access::sample_grpc::v1::digital_twin_consumer::digital_twin_consumer_server::DigitalTwinConsumerServer;
+use samples_protobuf_data_access::sample_grpc::v1::digital_twin_provider::digital_twin_provider_client::DigitalTwinProviderClient;
+use samples_protobuf_data_access::sample_grpc::v1::digital_twin_provider::{GetRequest, SetRequest};
+use serde_derive::{Deserialize, Serialize};
+use std::net::SocketAddr;
+use tokio::time::{sleep, Duration};
+use tonic::transport::Server;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Property {
+    #[serde(rename = "MassageAirbags")]
+    massage_airbags: sdv::airbag_seat_massager::massage_airbags::TYPE,
+    #[serde(rename = "$metadata")]
+    metadata: Metadata,
+}
+
+/// Start the seat massage sequence.
+///
+/// # Arguments
+/// `provider_uri` - The provider uri.
+fn start_seat_massage_sequence(provider_uri: String) {
+    debug!("Starting the consumer's seat massage sequence.");
+
+    let mut crest_row: i8 = 0;
```

Comment:
what are these concepts of a row, crest, and wave? is this documented somewhere?

Reply:
This is just a sample.  The airbag sequence mimics a wave motion.  I have made some improvements to the code's documentation to help the reader get a better understanding.